### PR TITLE
Add hqmf_id to StatementReferences, Observations and Stratifications

### PR DESCRIPTION
**Pull request by hossenlopp**
Matching the changes in https://github.com/projecttacoma/cqm-models/pull/85. These fields will be needed for QRDA/cqm-reports.

Pull requests into cqm-converter require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

❗️ BEFORE MERGING
- [x] Update Gemfile to point to cqm-models master AFTER https://github.com/projecttacoma/cqm-models/pull/85 is merged.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/BONNIE-1921 https://jira.mitre.org/browse/BONNIE-1923
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] If applicable, the library version number in `cqm-converter.gemspec` has been updated N/A
- [x] Update cqm-execution test fixtures using update_cqm_execution_fixtures.sh script from server-scripts repo   PR: https://github.com/projecttacoma/cqm-execution/pull/31

**Bonnie Reviewer:**

Name: @daco101 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Cypress Reviewer:**

Name: @dczulada 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: lib/tasks/bonnie.rake [hunk 1]
```diff
@@ -36,8 +36,13 @@ namespace :cqm do
         else
           hds_value_sets = []
           value_set_fixtures.each_pair do |_, versions|
-            versions.each_pair do |_, vs_json|
-              hds_value_sets << HealthDataStandards::SVS::ValueSet.new(vs_json)
+            # if there is a _id then this level is a actually a valueset fixture
+            if versions.key?('_id')
```
Comment (daco101) [inline lib/tasks/bonnie.rake]: if I understand correctly, the values set fixture file can be
an array of valuesets
a version to oid to valueset map
a version to valueset map
Maybe add a comment about the supported formats?

File: lib/cqm/converter/bonnie_measure.rb [hunk 2]
```diff
@@ -147,6 +140,30 @@ def self.to_cqm(bonnie_measure)
       cqm_measure
     end
 
+    def self.convert_observations(bonnie_measure, main_cql_library)
+      observations = []
+      bonnie_measure&.observations&.each_with_index do |bonnie_observation, observation_index|
+        # if this happens to be a multiple observation measure _this is unlikely_ we need to make a key to grab the proper hqmf_id
+        observation_population_key = observation_index.positive? ? "OBSERV_#{observation_index - 1}" : 'OBSERV'
```
Comment (daco101) [inline lib/cqm/converter/bonnie_measure.rb]: Is there a test for this? I can't find `OBSERV_#{number}` in the repo. It looks like this is assuming a multiple observation measure has keys of "OBSERV", and also "OBSERV_0", is that correct?

Comment (hossenlopp) [reply lib/cqm/converter/bonnie_measure.rb]: We don't have any measures with multiple observations to test this code with. Multiple observation measures are possible but haven't shown themselves in the real world yet.

File: lib/tasks/bonnie.rake (hunk 1 above)
Comment (hossenlopp) [reply lib/tasks/bonnie.rake]: Yeah. There are now three different types of value set fixtures, it's very frustrating. I'll add a comment to explain that.

File: lib/cqm/converter/bonnie_measure.rb (hunk 2 above)
Comment (hossenlopp) [reply lib/cqm/converter/bonnie_measure.rb]: I have no clue why i did the `- 1` gonna remove that.

Comment (hossenlopp) [reply lib/cqm/converter/bonnie_measure.rb]: We won't be testing the support of conversion of multiple observation measures. This is because cqm-converters is a temporary project for our conversion and we already know we do not have to deal with multiple observation measures. A mongo query to check for this is:
```javascript
db.getCollection('cql_measures').find({ observations: {$exists: true}, $where: 'this.observations.length > 1'})
```

File: lib/cqm/converter/bonnie_measure.rb [hunk 3]
```diff
@@ -79,7 +79,8 @@ def self.to_cqm(bonnie_measure)
 
           population_map[population_name.to_sym] = CQM::StatementReference.new(
             library_name: cqm_measure.main_cql_library,
-            statement_name: get_cql_statement_for_bonnie_population_key(bonnie_measure.populations_cql_map, population_key)
+            statement_name: get_cql_statement_for_bonnie_population_key(bonnie_measure.populations_cql_map, population_key),
+            hqmf_id: bonnie_measure.population_criteria[population_key]['hqmf_id']
```
[the remark this answers is not in the document]
Comment (hossenlopp) [reply lib/cqm/converter/bonnie_measure.rb]: The `population_key` is from the value side of the this structure in the measure at `populations[n]`:
image

The multiple populations are handled since that key is used to reference more details about the populations in the `population_criteria` which looks like this:
image

[the remark this answers is not in the document]
Comment (dczulada) [reply lib/cqm/converter/bonnie_measure.rb]: got it.  and that population in is the HDS model.
